kasa/smartcam/modules/listen.py

```diff
@@ -0,0 +1,154 @@
+"""Implementation of motion detection module."""
+
+from __future__ import annotations
+
+import asyncio
+import logging
+import os
+import socket
+import sys
+import uuid
+from collections.abc import Callable, Iterable
+from datetime import timedelta
+from subprocess import check_output
+
+import onvif  # type: ignore[import-untyped]
+from aiohttp import web
+from onvif.managers import NotificationManager  # type: ignore[import-untyped]
+
+from ...credentials import Credentials
+from ...eventtype import EventType
+from ..smartcammodule import SmartCamModule
+
+_LOGGER = logging.getLogger(__name__)
+
+DEFAULT_LISTEN_PORT = 28002
+
+
+TOPIC_EVENT_TYPE = {
+    "tns1:RuleEngine/CellMotionDetector/Motion": EventType.MOTION_DETECTED,
+    "tns1:RuleEngine/CellMotionDetector/People": EventType.PERSON_DETECTED,
+    "tns1:RuleEngine/TamperDetector/Tamper": EventType.TAMPER_DETECTED,
+}
+
+
+class Listen(SmartCamModule):
+    """Implementation of lens mask module."""
+
+    manager: NotificationManager
+    callback: Callable[[EventType], None]
+    event_types: Iterable[EventType] | None
+    listening = False
+    site: web.TCPSite
+    runner: web.AppRunner
+    instance_id: str
+
+    async def _invoke_callback(self, event: EventType) -> None:
+        self.callback(event)
+
+    async def _handle_event(self, request: web.Request) -> web.Response:
+        content = await request.read()
+        result = self.manager.process(content)
+        for msg in result.NotificationMessage:
+            if (event := TOPIC_EVENT_TYPE.get(msg.Topic._value_1)) and (
```

Comment:
Worth logging all (or at least unknown?) incoming events to help adding new, currently unsupported ones in the future?

Reply:
Now logging all as above